README.md
# resources

[//]: # (automatically generated from https://github.com/metwork-framework/github_organization_management/blob/master/common_files/README.md)

**Status (master branch)**



[![Drone CI](http://metwork-framework.org:8000/api/badges/metwork-framework/resources/status.svg)](http://metwork-framework.org:8000/metwork-framework/resources)
[![Maintenance](https://github.com/metwork-framework/resources/blob/master/badges/maintained.svg)]()










## Contributing guide

See [CONTRIBUTING.md](CONTRIBUTING.md) file.



## Code of Conduct

See [CODE_OF_CONDUCT.md](CODE_OF_CONDUCT.md) file.



## Sponsors

*(If you are officially paid to work on MetWork Framework, please contact us to add your company logo here!)*

[![logo](https://raw.githubusercontent.com/metwork-framework/resources/master/sponsors/meteofrance-small.jpeg)](http://www.meteofrance.com)


CHANGELOG.md
# CHANGELOG

## [Unreleased]

### New Features

- add metwork_repos.py script
- add exact-level option
- add a cookiecutter template
- generate README.md file for integration_level >= 5
- first version of contributing guide
- better README
- add first drone support
- add mfcom support
- add rename support
- optional auth
- optional auth
- add a new fnmatch extension and use it for docker-mfxxx-yyy-buildimage_drone.yml
- generate .drone.yml and run_tests.sh from resource
- new templates to execute integration tests directly from modules
- add 0.5 support for mergify
- introduce mfextaddon
- put build logs on public website for mfextaddon_scientific
- add gitter badges
- add integration tests for mfext addons
- add MFEXT_ADDON_NAME in root.mk for mfext addons
- add new logos
- add png logos
- add logo
- prepare 0.7 release
- better run_integration_tests
- add favicon
- Python3 version is now 3.7
- add some changes from mfextaddon_scientific
- remove typo
- use new build cache
- add mf logo
- add small mf logo
- add sponsors
- add 0.8 version
- introduce test image
- introduce testimage
- new packaging
- new mfext bootstrap
- add installation guide
- add doc files
- replace MODULE* environment variables names by MFMODULE* (MODULE_HOME becomes MFMODULE_HOME and so on)
- we move to portable builds
- remove mfcom (now integrated into mfext)
- remove mfcom (now integrated into mfext)
- add cheatsheet link
- introduce 0.9
- add .gitignore and .drone.yml for python2_tests repositories
- update documentation

### Bug Fixes

- fix bad space
- yum install {{MODULE}} before call to {{MODULE}}_wrapper
- fix latest commit
- fix utf8 issues
- fix utf8 char
- directory integration_tests may no exist
- tag condition
- fix renaming feature in subdir
- some fixes about mfext addon drone conf
- publish addons in /pub
- remove doc publish for mfext addons
- don t try to execute tests on data directory
- PR CI on release branches
- install build dependency python3_scientific@mfext for mfextaddon_python3_ia
- fix integration tests for mfext addons
- fix tag for mfext addons
- add path to misc@mfext
- fix cheatsheet.md
- remove doc directory from repo docker-mfextaddon_python3_ia-centos7-buildimage
- force delete doc directory from repo docker-mfextaddon_python3_ia-centos7-buildimage
- typo
- add libform.so.5 as allowed extra dependencies
- fix integration tests for ci_* branches on mfext addons

## changelog_start (2018-06-12)

- No interesting change




cookiecutter/_{{cookiecutter.repo}}/mfxxx_integration_tests_runner_run_tests.sh
#!/bin/bash
# automatically generated from https://github.com/metwork-framework/resources/blob/master/cookiecutter/_%7B%7Bcookiecutter.repo%7D%7D/mfxxx_drone.yml template

{% if cookiecutter.repo|fnmatch('mf*-integration-tests-runner') %}
    {% set LIST = cookiecutter.repo.split("-") %}
    {% set MODULE = LIST[0] %}
{% else %}
    {% set MODULE = "" %}
{% endif %}

cd {{MODULE}}/integration_tests

if test $? != 0; then
    echo "Directory integration_tests is missing"
    exit 0
fi

list_rep=`ls`
if test -z "$list_rep"; then
    echo "There are no integration tests"
    exit 0
fi

ret=0
for rep in $list_rep; do
    if [ -d $rep ]; then
        cd $rep
        if test -s .layerapi2_dependencies; then
             LAYERS_TO_LOAD=`cat .layerapi2_dependencies |xargs |sed 's/ /,/g'` 
             WRAPPER=0
        else
             WRAPPER=1
        fi
        for test in test*.sh; do
            echo "Test" $test "in" $rep
            if test $WRAPPER -eq 0; then
                layer_wrapper --layers=$LAYERS_TO_LOAD -- ./$test
            else
                ./$test
            fi
            if test $? == 0; then
                echo "Test $test OK"
            else
                echo "Test $test KO"
                ret=1
            fi
        done
        cd ..
    fi
done

exit $ret


cookiecutter/_{{cookiecutter.repo}}/integration_tests/0095_suspicious_dependencies/test.sh
#!/bin/bash

# find external dependencies (system) with a dynamic link with metwork libs

export PATH=${PATH}:${PWD}/..
RET=0
FIC_DEPS=`pwd`/deps
{% if "mfext-addon" not in "REPO_TOPICS"|getenv|from_json %}
cd "${MFMODULE_HOME}" || exit 1
external_dependencies.sh >${FIC_DEPS}
for F in $(cat ${FIC_DEPS}); do
    N=$(ldd "${F}" 2>/dev/null |grep -c metwork)
    if test "$N" -gt 0; then
        echo "***** $F *****"
        echo "=== ldd |grep metwork ==="
        ldd "${F}" |grep metwork
        echo
        echo "=== revert ldd ==="
        revert_ldd.sh "${F}"
        echo
        echo
        RET=1
    fi
done
{% else %}
cd "${MFEXT_HOME}" || exit 1
cd opt
for layer in `ls`; do
    cd "${layer}"
    if test -f .mfextaddon; then
        addon="mfextaddon_"`cat .mfextaddon`
        if test "{{cookiecutter.repo}}" = "${addon}"; then
            echo
            echo "=== Suspicious dependencies layer ${layer} ==="
            echo
            external_dependencies.sh >${FIC_DEPS}
            for F in $(cat ${FIC_DEPS}); do
                N=$(ldd "${F}" 2>/dev/null |grep -c metwork)
                if test "$N" -gt 0; then
                    echo "***** $F *****"
                    echo "=== ldd |grep metwork ==="
                    ldd "${F}" |grep metwork
                    echo
                    echo "=== revert ldd layer ${layer} ==="
                    revert_ldd.sh "${F}"
                    echo
                    echo
                    RET=1
                fi
            done
        fi
    fi
    cd ..
done


{% endif %}

rm -f ${FIC_DEPS}
if test "${RET}" = "1"; then
    echo "suspicious dependencies found"
    #exit 1
fi


cookiecutter/_{{cookiecutter.repo}}/integration_tests/0096_system_extra_dependencies/test.sh
#!/bin/bash


OK_DEPS=$(cat list.txt list_ok_not_found.txt |xargs)

N=$(cat /etc/redhat-release 2>/dev/null |grep -c "^CentOS release 6")
if test "${N}" -eq 0; then
    echo "We test this only on centos6"
    exit 0
fi

export PATH=${PATH}:${PWD}/..
RET=0

{% if "mfext-addon" not in "REPO_TOPICS"|getenv|from_json %}
cd "${MFMODULE_HOME}" || exit 1
cd opt
for layer in `ls`; do
    cd "${layer}"
    current_layer=`cat .layerapi2_label`
    DEPS1=$(layer_wrapper --layers=${current_layer} -- external_dependencies.sh |awk -F '/' '{print $NF}' |xargs)
    DEPS2=$(layer_wrapper --layers=${current_layer} -- external_dependencies_not_found.sh |xargs)
    DEPS=$(echo $DEPS1 $DEPS2)
    for DEP in ${DEPS}; do
        FOUND=0
        for OK_DEP in ${OK_DEPS}; do
            if test "${DEP}" = "${OK_DEP}"; then
                FOUND=1
                break
            fi
        done
        if test "${FOUND}" = "1"; then
            continue
        fi
        echo "***** ${DEP} *****"
        echo "=== revert ldd ==="
        revert_ldd.sh "${DEP}"
        echo
        echo
        RET=1
    done
    cd ..
done
{% else %}
cd "${MFEXT_HOME}" || exit 1
cd opt
for layer in `ls`; do
    cd "${layer}"
    if test -f .mfextaddon; then
        addon="mfextaddon_"`cat .mfextaddon`
        if test "{{cookiecutter.repo}}" = "${addon}"; then
            echo
            echo "=== System extra dependencies layer ${layer} ==="
            echo
            current_layer=`cat .layerapi2_label`
            DEPS1=$(layer_wrapper --layers=${current_layer} -- external_dependencies.sh |awk -F '/' '{print $NF}' |xargs)
            DEPS2=$(layer_wrapper --layers=${current_layer} -- external_dependencies_not_found.sh |xargs)
            DEPS=$(echo $DEPS1 $DEPS2)
            for DEP in ${DEPS}; do
                FOUND=0
                for OK_DEP in ${OK_DEPS}; do
                    if test "${DEP}" = "${OK_DEP}"; then
                        FOUND=1
                        break
                    fi
                done
                if test "${FOUND}" = "1"; then
                    continue
                fi
                echo "***** ${DEP} *****"
                echo "=== revert ldd layer ${layer} ==="
                revert_ldd.sh "${DEP}"
                echo
                echo
                RET=1
            done
        fi
    fi
    cd ..
done
{% endif %}

if test "${RET}" = "1"; then
    echo "extra dependencies found"
    exit 1
fi


scripts/metwork_repos.py
#!/usr/bin/env python3

import argparse
import asyncio
from aiohttp import ClientSession, BasicAuth, ClientTimeout
import os
import aiohttp_github_helpers as h

GITHUB_USER = os.environ.get('GITHUB_USER', None)
GITHUB_PASS = os.environ.get('GITHUB_PASS', None)
TIMEOUT = ClientTimeout(total=20)
AUTH = None
if GITHUB_USER is not None and GITHUB_PASS is not None:
    AUTH = BasicAuth(GITHUB_USER, GITHUB_PASS)
ORG = "metwork-framework"
TOPICS_TO_EXCLUDE = ["testrepo"]


async def getrepos(minimal_level, exact_level):
    repos = []
    async with ClientSession(auth=AUTH, timeout=TIMEOUT) as session:
        if exact_level:
            max_level = minimal_level + 1
        else:
            max_level = 6
        for i in range(minimal_level, max_level):
            topics_to_include = ["integration-level-%i" % i]
            tmp = await h.github_get_org_repos_by_topic(session,
                                                        ORG,
                                                        topics_to_include,
                                                        TOPICS_TO_EXCLUDE)
            repos = repos + tmp
    return repos

parser = argparse.ArgumentParser(description='get metwork-framework repos')
parser.add_argument('--minimal-level', type=int, default=1,
                    help='minimal integration level')
parser.add_argument('--exact-level', action='store_true',
                    help='if set the minimal-level must be the exact level')
args = parser.parse_args()
loop = asyncio.get_event_loop()
reply = loop.run_until_complete(getrepos(args.minimal_level, args.exact_level))
loop.close()
for repo in reply:
    print(repo)


scripts/metwork_topics.py
#!/usr/bin/env python3

import argparse
import asyncio
import json
from aiohttp import ClientSession, BasicAuth, ClientTimeout
import os
import aiohttp_github_helpers as h

GITHUB_USER = os.environ.get('GITHUB_USER', None)
GITHUB_PASS = os.environ.get('GITHUB_PASS', None)
TIMEOUT = ClientTimeout(total=20)
AUTH = None
if GITHUB_USER is not None and GITHUB_PASS is not None:
    AUTH = BasicAuth(GITHUB_USER, GITHUB_PASS)
ORG = "metwork-framework"
TOPICS_TO_EXCLUDE = ["testrepo"]


async def get_repo_topics(owner, name):
    topics = []
    async with ClientSession(auth=AUTH, timeout=TIMEOUT) as session:
        topics = await h.github_get_repo_topics(session, owner, name)
    return topics

parser = argparse.ArgumentParser(description='get repo topics')
parser.add_argument('owner', type=str, help='repo owner')
parser.add_argument('name', type=str, help='repo name')
parser.add_argument('--json', action='store_true',
                    help='if set, format the output as a json list')
args = parser.parse_args()
loop = asyncio.get_event_loop()
reply = loop.run_until_complete(get_repo_topics(args.owner, args.name))
loop.close()
if args.json:
    print(json.dumps(reply))
else:
    for topic in reply:
        print(topic)


cookiecutter/hooks/post_gen_project.py
#!/usr/bin/env python3

import os
import shutil


def test_empty(path):
    try:
        with open(path, 'rb') as f:
            content = f.read().strip()
            if len(content) == 0:
                return True
    except Exception:
        pass
    return False


paths_to_delete = []
for path in [os.path.join(x[0], y) for x in os.walk('.') for y in x[2]]:
    if path.endswith('.rename'):
        continue
    if test_empty(path):
        paths_to_delete.append(path)
    if path.endswith('.delete'):
        if not test_empty(path):
            paths_to_delete.append(path.replace('.delete', ''))
            paths_to_delete.append(path)

for path in paths_to_delete:
    if os.path.basename(path) in ['__init__.py']:
        continue
    print("remove %s" % path)
    try:
        os.unlink(path)
    except Exception:
        pass
    try:
        shutil.rmtree(path, ignore_errors=True)
    except Exception:
        pass


paths_to_rename = []
paths_to_delete = []
for path in [os.path.join(x[0], y) for x in os.walk('.') for y in x[2]]:
    if path.endswith('.rename'):
        if test_empty(path):
            paths_to_delete.append(path)
            paths_to_delete.append(path.replace('.rename', ''))
            continue
        with open(path, 'rb') as f:
            content = f.read().strip().decode("utf8")
            if len(content.encode('utf8').split(b"\n")) != 1:
                raise Exception("bad content: [%s] for path=%s" %
                                (content, path))
            newpath = os.path.join(os.path.dirname(path), content)
            paths_to_rename.append((path.replace('.rename', ''), newpath))
            paths_to_delete.append(path)

for source, target in paths_to_rename:
    print("move %s => %s" % (source, target))
    shutil.move(source, target)

for path in paths_to_delete:
    print("remove %s" % path)
    try:
        os.unlink(path)
    except Exception:
        pass
    try:
        shutil.rmtree(path, ignore_errors=True)
    except Exception:
        pass

if os.path.exists("README.md.improve") and os.path.exists("README.md"):
    print("improving README.md...")
    os.system("improve_readme.sh ./README.md")
try:
    os.unlink("README.md.improve")
except Exception:
    pass


scripts/metwork_valid_merge_logic_status.py
#!/usr/bin/env python3

import argparse
import asyncio
from aiohttp import ClientSession, BasicAuth, ClientTimeout
import os
import sys
import logging
import aiohttp_github_helpers as h

GITHUB_USER = os.environ['GITHUB_USER']
GITHUB_PASS = os.environ['GITHUB_PASS']
TIMEOUT = ClientTimeout(total=20)
AUTH = BasicAuth(GITHUB_USER, GITHUB_PASS)
ORG = "metwork-framework"


async def valid_status(owner, repo, sha):
    async with ClientSession(auth=AUTH, timeout=TIMEOUT) as session:
        r = await h.github_create_status(session, owner, repo, sha, "success",
                                         "http://metwork-framework.org:9000/"
                                         "github_webhook_ready_to_merge",
                                         "pr ready to merge logic",
                                         "pr ready to merge")
        if not(r):
            logging.critical("Impossible to create status for %s/%s@%s" %
                             (owner, repo, sha))
            sys.exit(1)
        print("status for %s/%s@%s: ok" % (owner, repo, sha))

parser = argparse.ArgumentParser(description='valid merge logic status')
parser.add_argument('REPO', type=str, help='repo name (without owner)')
parser.add_argument('SHA', type=str, help='sha to valid')

args = parser.parse_args()
loop = asyncio.get_event_loop()

loop.run_until_complete(valid_status(ORG, args.REPO, args.SHA))
loop.close()


cookiecutter/_{{cookiecutter.repo}}/mfxxx_bootstrap.sh
#!/bin/bash

{% set MODULE = cookiecutter.repo|upper %}

set -eu

function get_abs_filename() {
    echo "$(cd "$(dirname "$1")" && pwd)/$(basename "$1")"
}

function usage() {
    echo "usage: ./bootstrap.sh INSTALL_PREFIX_DIRECTORY MFEXT_INSTALL_ROOT_DIRECTORY"
}

if test "${1:-}" = "" -o "${2:-}" = ""; then
    usage
    exit 1
fi
if test "${1:-}" = "--help"; then
    usage
    exit 0
fi

MFEXT_HOME=$(get_abs_filename "$2")
export MFEXT_HOME
if ! test -d "${MFEXT_HOME}"; then
    usage
    echo "ERROR: ${MFEXT_HOME} is not a directory"
    exit 1
fi
MFEXT_HOME=$(get_abs_filename "${MFEXT_HOME}")
export MFEXT_HOME
MFEXT_VERSION=$(cat "${MFEXT_HOME}/config/version")
export MFEXT_VERSION
{{MODULE}}_VERSION=$("${MFEXT_HOME}/bin/guess_version.sh")
export {{MODULE}}_VERSION
MFMODULE_VERSION=$("${MFEXT_HOME}/bin/guess_version.sh")
export MFMODULE_VERSION

MFMODULE_HOME=$(get_abs_filename "$1")
export MFMODULE_HOME
if ! test -d "${MFMODULE_HOME}"; then
    usage
    echo "ERROR: ${MFMODULE_HOME} is not a directory"
    exit 1
fi

if ! test -f "${MFEXT_HOME}/bin/guess_version.sh"; then
    echo "ERROR: configured mfext home (${MFEXT_HOME}) is not a mfext home"
    exit 1
fi

export MFMODULE={{MODULE}}
export MFMODULE_LOWERCASE={{cookiecutter.repo}}
SRC_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
export SRC_DIR

export MODULE_HAS_HOME_DIR=1

rm -f adm/root.mk
touch adm/root.mk

ROOT_PATH=${MFEXT_HOME}/opt/misc/bin:${MFEXT_HOME}/bin:${MFEXT_HOME}/opt/core/bin:/usr/sbin:/usr/bin:/sbin:/bin
ROOT_LD_LIBRARY_PATH=""
ROOT_PKG_CONFIG_PATH=""
ROOT_LAYERAPI2_LAYERS_PATH=${MFMODULE_HOME}/opt:${MFMODULE_HOME}:${MFEXT_HOME}/opt:${MFEXT_HOME}

echo "Making adm/root.mk..."
rm -f adm/root.mk
touch adm/root.mk

echo "unexport MFMODULE_RUNTIME_HOME" >>adm/root.mk
echo "unexport MFMODULE_RUNTIME_SUFFIX" >>adm/root.mk
echo "unexport MFMODULE_RUNTIME_USER" >>adm/root.mk

echo "export MFMODULE := ${MFMODULE}" >>adm/root.mk
echo "export MFMODULE_LOWERCASE := $(echo ${MFMODULE} | tr '[:upper:]' '[:lower:]')" >>adm/root.mk
echo "export LAYERAPI2_LAYERS_PATH := ${ROOT_LAYERAPI2_LAYERS_PATH}" >>adm/root.mk
echo "export MFEXT_HOME := ${MFEXT_HOME}" >>adm/root.mk
echo "export MFEXT_VERSION := ${MFEXT_VERSION}" >>adm/root.mk
echo "export MFMODULE_HOME := ${MFMODULE_HOME}" >>adm/root.mk
echo "export MFMODULE_VERSION := {% raw %}${{% endraw %}{{MODULE}}_VERSION}" >>adm/root.mk
echo "export SRC_DIR := ${SRC_DIR}" >>adm/root.mk
echo "ifeq (\$(FORCED_PATHS),)" >>adm/root.mk
echo "  export PATH := ${ROOT_PATH}" >>adm/root.mk
echo "  export LD_LIBRARY_PATH := ${ROOT_LD_LIBRARY_PATH}" >>adm/root.mk
echo "  export PKG_CONFIG_PATH := ${ROOT_PKG_CONFIG_PATH}" >>adm/root.mk
echo "  LAYER_ENVS:=\$(shell env |grep '^LAYERAPI2_LAYER_.*_LOADED=1\$\$' |awk -F '=' '{print \$\$1;}')" >>adm/root.mk
echo "  \$(foreach LAYER_ENV, \$(LAYER_ENVS), \$(eval unexport \$(LAYER_ENV)))" >>adm/root.mk
echo "endif" >>adm/root.mk
echo "export ${MFMODULE}_HOME := ${MFMODULE_HOME}" >>adm/root.mk
echo "export ${MFMODULE}_VERSION := {% raw %}${{% endraw %}{{MODULE}}_VERSION}" >>adm/root.mk
if test "${MODULE_HAS_HOME_DIR:-}" = "1"; then
    echo "export MODULE_HAS_HOME_DIR := 1" >>adm/root.mk
fi
if test "${FTP_PROXY:-}" != ""; then
    echo "export FTP_PROXY:=${FTP_PROXY:-}" >>adm/root.mk
fi
if test "${http_proxy:-}" != ""; then
    echo "export http_proxy:=${http_proxy:-}" >>adm/root.mk
fi
if test "${https_proxy:-}" != ""; then
    echo "export https_proxy:=${https_proxy:-}" >>adm/root.mk
fi
if test "${HTTPS_PROXY:-}" != ""; then
    echo "export HTTPS_PROXY:=${HTTPS_PROXY:-}" >>adm/root.mk
fi
if test "${HTTP_PROXY:-}" != ""; then
    echo "export HTTP_PROXY:=${HTTP_PROXY:-}" >>adm/root.mk
fi
echo "export PYTHON2_SHORT_VERSION := 2.7" >>adm/root.mk
echo "export PYTHON3_SHORT_VERSION := 3.7" >>adm/root.mk

echo "BOOTSTRAP DONE !"
echo "MFEXT_HOME=${MFEXT_HOME}"


cookiecutter/_{{cookiecutter.repo}}/mfextaddon_bootstrap.sh
#!/bin/bash

{%- if "mfext-addon" in "REPO_TOPICS"|getenv|from_json %}
{%- set MFEXT_ADDON_NAME = cookiecutter.repo.split("_")[1:]|join("_") %}
{%- else %}
{%- set MFEXT_ADDON_NAME = "" %}
{%- endif %}

set -eu

function get_abs_filename() {
    echo "$(cd "$(dirname "$1")" && pwd)/$(basename "$1")"
}

function usage() {
   echo "usage: ./bootstrap.sh MFEXT_HOME"
}

if test "${1:-}" = "--help"; then
    usage
    exit 0
fi
if test "${1:-}" = ""; then
    usage
    exit 1
fi

MFEXT_HOME=$(get_abs_filename "$1")
export MFEXT_HOME
MFEXT_VERSION=$(cat "${MFEXT_HOME}/config/version")
export MFEXT_VERSION
MFMODULE_VERSION=$("${MFEXT_HOME}/bin/guess_version.sh")
export MFMODULE_VERSION
export MFMODULE_HOME=${MFEXT_HOME}

export MFMODULE=MFEXT
export MFMODULE_LOWERCASE=mfext
export MFEXTADDON=1
SRC_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
export SRC_DIR

if ! test -d adm; then mkdir adm; fi
rm -f adm/root.mk
touch adm/root.mk

ROOT_PATH=${MFEXT_HOME}/bin:${MFEXT_HOME}/opt/core/bin:${PATH:-}

echo "Making adm/root.mk..."
rm -f adm/root.mk adm/envtpl
touch adm/root.mk
ln -s "${MFEXT_HOME}/bin/envtpl" adm/envtpl

echo "export MFMODULE := ${MFMODULE}" >>adm/root.mk
echo "export MFMODULE_LOWERCASE := $(echo ${MFMODULE} | tr '[:upper:]' '[:lower:]')" >>adm/root.mk
echo "export LAYERAPI2_LAYERS_PATH := ${MFEXT_HOME}/opt:${MFEXT_HOME}" >>adm/root.mk
echo "export MFEXT_HOME := ${MFEXT_HOME}" >>adm/root.mk
echo "export MFEXT_ADDON := 1" >>adm/root.mk
echo "export MFEXT_ADDON_NAME := {{MFEXT_ADDON_NAME}}" >>adm/root.mk
echo "export MFEXT_VERSION := ${MFEXT_VERSION}" >>adm/root.mk
echo "export MFMODULE_HOME := ${MFMODULE_HOME}" >>adm/root.mk
echo "export MFMODULE_VERSION := ${MFEXT_VERSION}" >>adm/root.mk
echo "export SRC_DIR := ${SRC_DIR}" >>adm/root.mk
echo "ifeq (\$(FORCED_PATHS),)" >>adm/root.mk
echo "  export PATH := ${ROOT_PATH}" >>adm/root.mk
echo "  export LD_LIBRARY_PATH := ${MFEXT_HOME}/lib" >>adm/root.mk
echo "  export PKG_CONFIG_PATH := ${MFEXT_HOME}/lib/pkgconfig" >>adm/root.mk
echo "  LAYER_ENVS:=\$(shell env |grep '^LAYERAPI2_LAYER_.*_LOADED=1\$\$' |awk -F '=' '{print \$\$1;}')" >>adm/root.mk
echo "  \$(foreach LAYER_ENV, \$(LAYER_ENVS), \$(eval unexport \$(LAYER_ENV)))" >>adm/root.mk
echo "endif" >>adm/root.mk

if test "${ftp_proxy:-}" != ""; then
    echo "export ftp_proxy:=${ftp_proxy:-}" >>adm/root.mk
fi
if test "${FTP_PROXY:-}" != ""; then
    echo "export FTP_PROXY:=${FTP_PROXY:-}" >>adm/root.mk
fi
if test "${http_proxy:-}" != ""; then
    echo "export http_proxy:=${http_proxy:-}" >>adm/root.mk
fi
if test "${https_proxy:-}" != ""; then
    echo "export https_proxy:=${https_proxy:-}" >>adm/root.mk
fi
if test "${HTTPS_PROXY:-}" != ""; then
    echo "export HTTPS_PROXY:=${HTTPS_PROXY:-}" >>adm/root.mk
fi
if test "${HTTP_PROXY:-}" != ""; then
    echo "export HTTP_PROXY:=${HTTP_PROXY:-}" >>adm/root.mk
fi

# FIXME: do not hardcode this
# FIXME: move to layer root extra_env ?
echo "export PYTHON2_SHORT_VERSION := 2.7" >>adm/root.mk
echo "export PYTHON3_SHORT_VERSION := 3.7" >>adm/root.mk

echo "BOOTSTRAP DONE !"
echo "MFEXT_HOME=${MFEXT_HOME}"
